Quicksearch: use class icons when available

NiklasBr opened the pull request:
## Changes in this pull request  
Resolves #9498 which broke during rebase.

## Additional info

NiklasBr committed:
Class icon in SearchController

NiklasBr committed:
Show custom class icon in startup.js

NiklasBr committed:
ExtJs tpl syntax

NiklasBr committed:
Use addAdminStyle trait for event dispatch

NiklasBr committed:
Use AdminStyleTrait

brusch commented on bundles/AdminBundle/Controller/Searchadmin/SearchController.php:
```diff
@@ -478,14 +481,25 @@ public function quicksearchAction(Request $request, EventDispatcherInterface $ev
+                    'icon' => '',
                 ];
 
                 if ($element instanceof Asset) {
```
I think we don't need that whole code block at all (incl. for Assets), because I think that should be now covered by `$this->addAdminStyle($element, null, $data);`. See also: https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/models/Element/AdminStyle.php#L61

NiklasBr replied:
In that case I will need to refactor how classes are treated for `variants` where the parent object has a custom icon since one is added by a `class` and the other as a an image `src`: 
![image](https://user-images.githubusercontent.com/279826/127129772-89da9cfb-3031-4eea-b008-e7f6657bf1c4.png)

brusch commented on bundles/AdminBundle/Controller/Searchadmin/SearchController.php:
```diff
@@ -478,14 +481,25 @@ public function quicksearchAction(Request $request, EventDispatcherInterface $ev
                     $data['iconCls'] .= ' pimcore_icon_' . $element->getType();
                 }
 
+                $this->addAdminStyle($element, null, $data);
```
I think we should introduce a new context for the search `\Pimcore\Event\Admin\ElementAdminStyleEvent::CONTEXT_SEARCH` instead of passing `null`.

brusch commented:
@NiklasBr 🏓Thanks 😉

NiklasBr committed:
Adjustments after PR review:
 - ElementAdminStyleEvent constant
 - Styling of icon
 - Changed logic for handling class/icon conflict

NiklasBr committed:
Merge branch '10.x' into quicksearch-icons

NiklasBr committed:
Fix after conflict merge

brusch commented:
Excellent, thanks a lot! 👍
